packages/ag-charts-community/src/util/json.test.ts

```diff
@@ -526,369 +526,761 @@ describe('json module', () => {
+                    d: {
+                        f: {
+                            $if: [
+                                {
+                                    $or: [
+                                        { $eq: [{ $path: '../a' }, 'cousin'] },
+                                        { $eq: [{ $path: '../a' }, 'parent'] },
+                                    ],
+                                },
+                                { $ref: 'key' },
+                                'no',
+                            ],
                         },
-                        '#000000',
-                        0.5,
-                    ],
-                },
-            };
-            jsonResolveOperations(source, {});
-            expect(source).toEqual({
-                a: {
-                    type: 'gradient',
-                    colorStops: [{ color: '#808080' }, { color: '#808000', stop: 0.3 }, { color: '#800000' }],
-                },
+                    },
+                };
+                const params = { key: 'hello', indirectRelative: { $ref: 'relative' }, relative: 'parent' };
+                const resolved = jsonResolveOperations([source, logic], params);
+                expect(resolved).toEqual({
+                    a: 'parent',
+                    b: { c: 'cousin' },
+                    d: { e: 'sibling', f: 'hello' },
+                });
             });
-        });
 
-        // Combined
-        it('should resolve nested operations', () => {
-            const source = {
-                a: { $ref: 'indirectRelative' },
-                b: {
-                    c: 'cousin',
-                },
-                d: {
-                    e: 'sibling',
-                    f: {
-                        $if: [
-                            { $or: [{ $eq: [{ $path: '../a' }, 'cousin'] }, { $eq: [{ $path: '../a' }, 'parent'] }] },
-                            { $ref: 'key' },
-                            'no',
+            it('should resolve nested operations with logical operations', () => {
+                const source = { a: { b: 'yes' } };
+                const logic = {
+                    a: {
+                        b: 'no',
+                        c: {
+                            $if: [
+                                {
+                                    $or: [{ $eq: [{ $path: './b' }, 'yes'] }, { $eq: [{ $path: './b' }, 'maybe'] }],
+                                },
+                                { $ref: 'one' },
+                                { $ref: 'two' },
+                            ],
+                        },
+                    },
+                };
+                const params = { one: 'ref-one', two: 'ref-two' };
+                const resolved = jsonResolveOperations([source, logic], params);
+                expect(resolved).toEqual({ a: { b: 'yes', c: 'ref-one' } });
+            });
+
+            it('should resolve options with defaults by type', () => {
```

Comment:
This test demonstrates the new `$apply` operation being used on chart-like options.